src/main/java/org/jabref/gui/util/DefaultTaskExecutor.java

```diff
@@ -23,21 +23,35 @@
 
     private static final Log LOGGER = LogFactory.getLog(DefaultTaskExecutor.class);
 
-    private ExecutorService executor = Executors.newFixedThreadPool(5);
+    private final ExecutorService executor = Executors.newFixedThreadPool(5);
 
     public static <V> V runInJavaFXThread(Callable<V> callable) {
-        FutureTask<V> task = new FutureTask<>(callable);
-        Platform.runLater(task);
-        try {
-            return task.get();
-        } catch (InterruptedException | ExecutionException e) {
-            LOGGER.error(e);
-            return null;
+        if (!Platform.isFxApplicationThread()) {
+            FutureTask<V> task = new FutureTask<>(callable);
+            Platform.runLater(task);
+            try {
+                return task.get();
+            } catch (InterruptedException | ExecutionException e) {
```

Comment:
You need a guarded wait here in case an `InterruptedException` is thrown

Reply:
I am not sure what you mean: The get already waits for completion. the only thing which we could add is a timeout. But that depends